app/assets/javascripts/tictactoe.js

```diff
@@ -1 +1,217 @@
 // Code your JavaScript / jQuery solution here
+// use 'odd' or 'even' to determine if it's X or O turn
+var turn = 0
+
+// const squares = [square_1,square_2,square_3,square_4,square_5,square_6,square_7,square_8,square_9]
+
+function isEven(n) {
+   return n % 2 === 0;
+}
+
+// Returns the token of the player whose turn it is, 'X' when the turn variable is even and 'O' when it is odd.
+function player() {
+  if (isEven(turn)) {
+    return 'X'
+  }
+  else {
+    return "O"
+  }
+}
+
+// Invokes player() and adds the returned string ('X' or 'O') to the clicked square on the game board.
+
+function updateState(square) {
+    if ($(square).text() == "") {
+      $(square).text(player())
+      turn++
+    }
+}
+
+// Accepts a string and adds it to the div#message element in the DOM.
+function setMessage(string) {
+  $('#message').text(string)
+}
+
+// 1) Returns true if the current board contains any winning combinations (three X or O tokens in a row, vertically, horizontally, or diagonally).
+//    Otherwise, returns false.
+// 2) If there is a winning combination on the board, checkWinner() should invoke setMessage(),
+//    passing in the appropriate string based on who won: 'Player X Won!' or 'Player O Won!'
+
+function checkWinner() {
+
+  var square_1 = $('td[data-x="0"][data-y="0"]').text()
+  var square_2 = $('td[data-x="1"][data-y="0"]').text()
+  var square_3 = $('td[data-x="2"][data-y="0"]').text()
+  var square_4 = $('td[data-x="0"][data-y="1"]').text()
+  var square_5 = $('td[data-x="1"][data-y="1"]').text()
+  var square_6 = $('td[data-x="2"][data-y="1"]').text()
+  var square_7 = $('td[data-x="0"][data-y="2"]').text()
+  var square_8 = $('td[data-x="1"][data-y="2"]').text()
+  var square_9 = $('td[data-x="2"][data-y="2"]').text()
+
+  var combination_1 = (square_1 == square_2 && square_1 == square_3 && square_1 != "") // X - X - X 1st row
```

Comment:
It might make more sense to store winning combinations as an array and iterate over that:

```
const WINNING_COMBOS = [[0,1,2], [3,4,5], [6,7,8], [0,3,6],
                        [1,4,7], [2,5,8], [0,4,8], [2,4,6]];
```